### ISSUE TITLE
Wholesale Pricing not sticking on orders placed in Admin

### ISSUE THREAD

**Issue by InternetMedicineMan** [magento/magento2#22252]
### Preconditions (*)
1. Magento 2.2.8 (upgraded from 2.2.7)
2. Multiple groups and group pricing assigned to customer levels

### Steps to reproduce (*)
1. From the admin create an order for wholesale user on wholesale website
2. Enter in products all pricing at price levels show up properly while creating
3. Save the order 

### Expected result (*)
1. Expect the wholesale order to have the pricing shown and defined for that price level

### Actual result (*)
1. Prices all get saved and charged at full retail

We were hoping that a module conflict was occurring from updating to 2.2.8 however, we've disabled all modules and it's still doing the same thing. I can create an admin order and the price will display at $300 for that customer level, and as soon as I save the order it shows up as $600 and looking at the credit card processing it charged $600.

This does not affect the front end at all, so far we've been able to get around it by just logging in as the customer and placing orders that way, which is less than ideal.

**Comment by Mobecls** [comment]
@InternetMedicineMan  - wasn't able to reproduce an issue on Magento 2.3-develop version. 

Case 1. 
- created a user. 
- added user to a wholesale group
- set a group price for wholesale groups to 90 (regular price was 100) on a product level.
- created order in backend for that user
- when order was saved - the price was set to 90.

Case 2. 
- created a new website "wholesale" (store and storeview too)
- set a group price for wholesale store to 90 (regular price was 100) on a product level.
- created order in backend for the user in "wholesale" website
- when the order was saved - the price was set to 90.

**Comment by InternetMedicineMan** [comment]
@mobecls I submitted the ticket for 2.2.8. But that’s great news it doesn’t occur in 2.3-develop

**Comment by ghost** [comment]
Hi @InternetMedicineMan thank you for your report, i'm not able to reproduce following steps you described on clean 2.2.-develop branch,, seems like this may be related to your custom code or modules.

**Comment by InternetMedicineMan** [comment]
Strange I removed all custom code and modules per my ticket, running just a plain install but with my data. The upgrade must have ruined the data then, go figure.

**Comment by InternetMedicineMan** [comment]
I found the problem. It's a problem that seems to be related to customer groups when placing admin orders. The group would always reset to "General" for me. No mater how many times I set it to a wholesale price level. I would see wholesale pricing but when saving it would be retail pricing.

The issue was reported and resolved with this patch:
https://github.com/magento/magento2/commit/d77bc6acaccaecddc4bda3415d268ed21ef94981

It's coming downline in version 2.2.9 which is why I'm experiencing in 2.2.8.

Just leaving a note in case anyone else catches this ticket as a similar issue.
